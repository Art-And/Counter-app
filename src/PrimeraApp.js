import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ useTest, subtitle }) => {

  return (
    <>
      <h1>{ useTest }</h1>
      <p>{ subtitle }</p>
    </>
  );
};

PrimeraApp.propTypes = {
  useTest: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  subtitle: 'Soy un test',
};

export default PrimeraApp;
