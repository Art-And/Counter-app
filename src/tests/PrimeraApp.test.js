import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {
  test('Debe mostrar <PrimeraApp /> correctamente', () => {

    const saludo = 'Hola soy una prueba';
    const wrapper = shallow(<PrimeraApp useTest={ saludo } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    const saludo = 'Hola soy una prueba';
    const subtitulo = 'Hola soy un subtitulo';

    const wrapper = shallow(
      <PrimeraApp
        useTest={ saludo }
        subtitle={ subtitulo }
      />);

    const textoParrafo = wrapper.find('p').text();
    console.log(textoParrafo);

    expect( textoParrafo ).toBe( subtitulo );

  });

});

