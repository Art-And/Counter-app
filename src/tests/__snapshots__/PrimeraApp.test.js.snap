// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy una prueba
    !!!
  </h1>
  <p>
    Soy un test
  </p>
</Fragment>
`;
